cilium/CFP-26135-default-egress-allow.md

```diff
@@ -0,0 +1,198 @@
+
+## Goals
+
+- Block egress traffic to certain endpoints without blocking all
+- Behavior of existing network policy or cilium network policy should not be affected
+
+## Non Goals
+ - Ingress Restriction without blocking all
+ - Applying this policy for specific endpoints
+ - Complete Network Policy support with all features
+
+## Proposal
+
+### Overview
+The proposal will talk about preferred option and also about other options that aims to solve the problem by allowing cloud provider to specify the intent of blocking egress to certain endpoints without blocking all. These deny rules can be plumbed either via iptables or bpf policy rules. IPtables rules would not work with ebpf host routing mode. This proposal considers to solve this via bpf policy rules as it will work for both legacy host routing and ebpf host routing modes. It will also ensure that the behavior of existing network policies or Cilium network policies is not regressed.
+
+### Option 1 (CommandLine Flag)
+The decision to pass input as command line was taken based on what is more appropriate for this scenario. There is a section below that describes the reason for going with this option.
+ 
+A new command line flag, --deny-tuples=[ip:port:protocol] could be added as stringSlice. Cloud providers would customize this cilium config to specify apriori that traffic from Cilium-managed Pods towards specific address+port+protocol combinations should be blocked. This flag would effectively act independently of all of the NP, CNP, CCNP resources and enforce a deny for such tuples regardless of the user-defined policy. The deny tuples from cilium config are sanitized and converted to cilium egress deny policy rules. Then these rules should get added to policy map of all endpoints. Several options are considered to implement this and the following section would talk about each option
+
+
+ * Option 1 is preferred as it aligns with existing code flow and does not affect existing network policies schema or behavior in any way.
+
+#### Adding new field in ruleMetadata
+Based on config received via command line, Cilium Agent would convert it to EgressDeny rule and does policyAdd. PolicyAdd will add the rules to repository and triggers policyreaction event to trigger identity allocation for labels specified in the policy. Cilium by default goes to deny mode if any policy added. To prevent this, this rule can be created with this flag `enableDefaultAllow` enabled. If a rule has this flag enabled, then cilium will never enforce default deny for the rest of traffic unless if there any egress k8s network policy or cilium network policy rule is enforced. By default, this flag is disabled and it would not regress existing network policy scenarios.
```

Comment:
> By default, this flag is disabled and it would not regress existing network policy scenarios.

This is a good goal, is there any planned tests to confirm this or do we think the existing testing coverage should give us reliable feedback about this? I guess the existing network policy tests should continue to work even if the new option is enabled, so that would be key to demonstrating this property.

Reply:
yes will create a separate test with this option enabled and existing network policy should continue to work.